action.yml

```diff
@@ -186,7 +186,12 @@ runs:
       uses: actions/cache/restore@v3
       with:
         path: |
-          ~/.stack
+          ${{ startsWith(runner.os, 'Windows') && 
+              '
+              ~\AppData\Roaming\stack
+              ~\AppData\Local\Programs\stack'
+            || '~/.stack'
+          }}
```

Comment:
I think it would be more consistent to do this in `setup` and expose a new `output` -- just like `setup.outputs.stack-works` works. This would also prevent bugs, since you'd be doing it in one place.